### PR TITLE
 Mushrooms: Rename spore nodes. Single ABM for mushroom growing

### PR DESCRIPTION
**Pull request by paramat**
This commit is the second half of approved #607 which is being split into 2 new commits.

### CODE REVIEW

File: mods/flowers/init.lua [hunk 1]
```diff
@@ -101,15 +160,15 @@ for _, m in pairs(mushrooms_datas) do
 		}
 	})
 
-	-- Register spores
+	-- Register mushroom spores
 
-	minetest.register_node("flowers:spores_" .. name, {
+	minetest.register_node("flowers:mushroom_spores_" .. name, {
 		description = string.sub(string.upper(name), 0, 1) ..
 			string.sub(name, 2) .. " Mushroom Spores",
```
Comment (kilbith) [inline mods/flowers/init.lua:167]: FFS, will you complete that clean-up by replacing that to `name:gsub("%l", string.upper, 1)` ? Same for the mushrooms.

Comment (paramat) [inline mods/flowers/init.lua]: Calm down, i copied your suggestion correctly and it didn't work, your previous suggestion didn't work either when correctly copied, please test your suggestions first.

**Comment by paramat** [comment]
47c7b0b187da669e1caacd8264825d29035eee6d

File: mods/flowers/init.lua (hunk 1 above)
Comment (kilbith) [inline mods/flowers/init.lua:167]: I actually tested and it worked fine. So whether you're lying or you should clean your glasses (the 1st time you confused the letter 'l' with the '1').

BTW, I always check my changes in-depth, not like you, considering the shitload of bugs we reported.

Comment (kilbith) [inline mods/flowers/init.lua:167]: As I'm unfairly accusated to do not test my suggestions... Here's some evidence : 

2015-08-07-202656_1280x800_scrot

2015-08-07-202659_1280x800_scrot

**Pull request closed without merging**